### PR TITLE
use CSS instead of placeholder labels to join fields together

### LINKED ISSUE #198: Grouped fields have not enough space between them for border
See 
image

Wee need to add a bit more spacing between grouped fields so the border (when not white) has enough place to be shown

### PR DESCRIPTION
**Pull request by kaligrafy**
also update demo survey to join all address fields together, closes #198

### CODE REVIEW

File: packages/evolution-legacy/src/styles/survey/_question.scss [hunk 1]
```diff
@@ -46,6 +46,29 @@ label strong, .apptr__form__label-standalone strong {
 
 }
 
+.apptr__form-join-next {
```
Comment (tahini) [inline packages/evolution-legacy/src/styles/survey/_question.scss:49]: You just have the `next` styes, will it work with previous too?

Comment (kaligrafy) [reply packages/evolution-legacy/src/styles/survey/_question.scss:49]: Unfortunatly, it is not possible to select backward adjacent divs in css... Can only do it forward... Should the joinWith only work forward?

Comment (kaligrafy) [reply packages/evolution-legacy/src/styles/survey/_question.scss:49]: But we can fix this by checking for previous AND next widget and convert all joinWith to next beforehand. Fixed!

File: packages/evolution-legacy/src/components/survey/Section.js [hunk 2]
```diff
@@ -167,14 +167,13 @@ export class Section extends React.Component {
...
-        const previousStatus = this.props.widgets[i - 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i - 1]}`, {}) : undefined;
-        const next = this.props.widgets[i + 1];
-        const nextStatus = this.props.widgets[i + 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i + 1]}`, {}) : undefined;
-
-        join = widgetConfig.joinWith === previous && previousStatus.isVisible ? 'previous' : widgetConfig.joinWith === next && nextStatus.isVisible? 'next' : undefined;
+      const nextWidgetShortname = this.props.widgets[i + 1];
+      const nextWidgetConfig = this.props.surveyContext.widgets[nextWidgetShortname];
+      const nextWidgetStatus = nextWidgetShortname ? _get(this.props.interview, `widgets.${nextWidgetShortname}`, {}) : undefined;
+      if (nextWidgetStatus && ((nextWidgetConfig.joinWith === widgetShortname && nextWidgetStatus.isVisible) || (widgetConfig.joinWith === nextWidgetShortname && nextWidgetStatus.isVisible))) {
+        join = 'next';
```
Comment (tahini) [inline packages/evolution-legacy/src/components/survey/Section.js:176]: There's only one possible choice for join: next, so you might as well use a boolean called join with next

File: packages/evolution-frontend/src/components/survey/Question.tsx [hunk 3]
```diff
@@ -329,9 +329,9 @@ export class Question<CustomSurvey, CustomHousehold, CustomHome, CustomPerson> e
         };
         const content = [(
             <div
-                key={"content_" + this.props.path}
+                key={'content_' + this.props.path}
                 style={{ position: 'relative' }}
-                className={`apptr__form-container${twoColumns ? ' two-columns' : ''}${
+                className={`apptr__form-container apptr__form-join-${this.props.join || 'none'}${twoColumns ? ' two-columns' : ''}${
```
Comment (tahini) [inline packages/evolution-frontend/src/components/survey/Question.tsx:334]: Is there a use case for the style `apptr__form-join-none`? Why not just not add a style if there is no join?

Also, now that we don't push/unshift an element to the content array, the content does not need to be an array anymore.

File: packages/evolution-frontend/src/components/survey/__tests__/__snapshots__/Question.test.tsx.snap [hunk 4]
```diff
@@ -1566,140 +1566,98 @@ exports[`With help popup and link Test widget with help 1`] = `
...
+        }
       }
+      type="text"
+      value=""
+    />
+  </div>
+</div>
+`;
+
+exports[`With joining questions Joining previous 1`] = `
```
Comment (tahini) [inline packages/evolution-frontend/src/components/survey/__tests__/__snapshots__/Question.test.tsx.snap:1617]: Now that we only join with next, this second test is not necessary anymore

**Review by tahini — APPROVED** [review]
Some comments, that can be fixed now, or later. If later, please open an issue to revisit the 'join' with all the comments in this PR.

File: packages/evolution-frontend/src/components/survey/Question.tsx (hunk 3 above)
Comment (kaligrafy) [reply packages/evolution-frontend/src/components/survey/Question.tsx:334]: Where is the push/unshift in the code?

Comment (tahini) [reply packages/evolution-frontend/src/components/survey/Question.tsx:334]: You removed it

File: packages/evolution-frontend/src/components/survey/__tests__/__snapshots__/Question.test.tsx.snap (hunk 4 above)
Comment (tahini) [reply packages/evolution-frontend/src/components/survey/__tests__/__snapshots__/Question.test.tsx.snap:1617]: Did you push the branch where you removed it?

Comment (kaligrafy) [reply packages/evolution-frontend/src/components/survey/__tests__/__snapshots__/Question.test.tsx.snap:1617]: yes, should be fine now

**Review by tahini — APPROVED** [review]

**Pull request merged**